### PR TITLE
Power section pages using collections

### PR DESCRIPTION
**Pull request by Deokumarjnu**
Fixes  https://github.com/quintype/ace-planning/issues/110

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: app/server/load-data.js [hunk 1]
```diff
@@ -47,14 +49,17 @@ export function loadErrorData(error, config) {
...
     switch (pageType) {
       case PAGE_TYPE.HOME_PAGE:
         return loadHomePageData(client, config, params.collectionSlug);
       case PAGE_TYPE.HOME_PREVIEW:
         return loadHomePageData(client, config);
       case PAGE_TYPE.SECTION_PAGE:
-        return loadSectionPageData(client, params.sectionId, config);
+        return loadSectionPageData(client, params.sectionId, config, publisherAttributes);
+      case PAGE_TYPE.COLLECTION_PAGE:
+        return loadCollectionPageData(client, params.collectionSlug, config);
```
Comment (Deokumarjnu) [inline app/server/load-data.js:62]: can we use common util function to fetch collection data for home and collection page ?

File: app/server/data-loaders/section-page-data.js [hunk 2]
```diff
@@ -1,15 +1,41 @@
...
+  const section = config.sections.find(section => section.id === sectionId) || {};
+  const sectionSlug = section.collection === null ? null : section.collection.slug;
+  if (sectionSlug && publisherAttributes.should_use_collection) {
+    return Collection.getCollectionBySlug(client, sectionSlug, { limit: 20 }, { depth: 2 }).then(collection => {
+      return {
+        section: section,
+        collection: collection.asJson(),
+        cacheKeys: collection.cacheKeys(config["publisher-id"])
+      };
+    });
```
Comment (Deokumarjnu) [inline app/server/data-loaders/section-page-data.js:14]: move this logic to utils, use async and await.

File: app/server/data-loaders/section-page-data.js [hunk 3]
```diff
@@ -1,15 +1,41 @@
...
-    section: config.sections.find(section => section.id === sectionId),
-    stories: stories.map(story => story.asJson()),
-    cacheKeys: [sorterToCacheKey(config["publisher-id"], "top", sectionId)].concat(
-      stories.map(story => storyToCacheKey(config["publisher-id"], story))
-    )
-  }));
+export function loadSectionPageData(client, sectionId, config, publisherAttributes) {
+  const section = config.sections.find(section => section.id === sectionId) || {};
+  const sectionSlug = section.collection === null ? null : section.collection.slug;
+  if (sectionSlug && publisherAttributes.should_use_collection) {
```
Comment (Deokumarjnu) [inline app/server/data-loaders/section-page-data.js:7]: move this condition to variables

File: app/server/data-loaders/section-page-data.js [hunk 4]
```diff
@@ -1,15 +1,41 @@
...
+    return Collection.getCollectionBySlug(client, sectionSlug, { limit: 20 }, { depth: 2 }).then(collection => {
+      return {
+        section: section,
+        collection: collection.asJson(),
+        cacheKeys: collection.cacheKeys(config["publisher-id"])
+      };
+    });
+  } else {
+    const storyFields =
+      "headline,subheadline,summary,sections,tags,author-name,author-id,authors,updated-at,last-published-at,published-at,updated-at,first-published-at,hero-image-metadata,hero-image-s3-key,story-content-id,slug,id,seo,story-template,metadata,url";
```
Comment (Deokumarjnu) [inline app/server/data-loaders/section-page-data.js:17]: move this to constant.js file

File: app/server/data-loaders/author-page-data.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { Author } from "@quintype/framework/server/api-client";
+import get from "lodash/get";
+
+export function loadAuthorPageData(client, authorSlug, config) {
+  let authorDetails = {};
+  const storyFields =
+    "alternative,slug,metadata,story-template,story-content-id,id,headline,hero-image-s3-key,hero-image-metadata,sections,tags,author-name,author-id,authors,created-at,first-published-at,published-at,last-published-at,url";
```
Comment (Deokumarjnu) [inline app/server/data-loaders/author-page-data.js:7]: move this to constant.js file and use limited field which is being used in cards

File: app/server/data-loaders/author-page-data.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  const params = { author: authorSlug, sort: "published-at", fields: storyFields, limit: 100 };
```
Comment (Deokumarjnu) [inline app/server/data-loaders/author-page-data.js:8]: use limit as 20

File: app/isomorphic/components/story-grid.js [hunk 7]
```diff
@@ -40,10 +40,10 @@ StoryGridStoryItem.propTypes = {
   position: PT.integer
 };
 
-export function StoryGrid(props) {
+export function StoryGrid({ stories = [] }) {
   return (
     <div className="story-grid">
```
Comment (Deokumarjnu) [inline app/isomorphic/components/story-grid.js:45]: return null if no stories

File: app/isomorphic/components/pages/section.js [hunk 8]
```diff
@@ -1,15 +1,33 @@
 import React from "react";
-
+import PT from "prop-types";
+import get from "lodash/get";
 import { StoryGrid } from "../story-grid";
+import { collectionToStories, LazyCollection } from "@quintype/components";
+import { getCollectionTemplate } from "../get-collection-template";
```
Comment (Deokumarjnu) [inline app/isomorphic/components/pages/section.js:6]: need spacing

File: app/isomorphic/components/pages/section.js [hunk 9]
```diff
@@ -1,15 +1,33 @@
 import React from "react";
-
+import PT from "prop-types";
```
Comment (Deokumarjnu) [inline app/isomorphic/components/pages/section.js:2]: use only needed proptype

File: app/isomorphic/components/pages/section.js [hunk 10]
```diff
@@ -1,15 +1,33 @@
...
+import get from "lodash/get";
 import { StoryGrid } from "../story-grid";
+import { collectionToStories, LazyCollection } from "@quintype/components";
+import { getCollectionTemplate } from "../get-collection-template";
+const SectionPage = props => {
+  const stories = collectionToStories(props.data.collection) || [];
+  const childCollections = get(props, ["data", "collection", "items"], []).filter(
```
Comment (Deokumarjnu) [inline app/isomorphic/components/pages/section.js:9]: check for null values `get` supports or not and handle it accordingly

File: app/isomorphic/components/pages/section.js [hunk 11]
```diff
@@ -1,15 +1,33 @@
...
+    item => item.type === "collection" && item.items.length > 0
+  );
+  const headline =
```
Comment (Deokumarjnu) [inline app/isomorphic/components/pages/section.js:12]: diff name ?

File: app/isomorphic/components/pages/collection-page.js [hunk 12]
```diff
@@ -0,0 +1,6 @@
+import React from "react";
+import { SectionPage } from "./section";
+
+export const CollectionPage = props => {
```
Comment (Deokumarjnu) [inline app/isomorphic/components/pages/collection-page.js:4]: remove { and return directly

File: app/isomorphic/components/pages/author-page/index.js [hunk 13]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import PT from "prop-types";
```
Comment (Deokumarjnu) [inline app/isomorphic/components/pages/author-page/index.js:2]: ? and space

File: app/isomorphic/components/pages/author-page/index.js [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+import { StoryGrid } from "../../story-grid";
+const AuthorPage = props => {
+  const stories = props.data.stories.map(({ story }) => story) || props.data.stories;
```
Comment (Deokumarjnu) [inline app/isomorphic/components/pages/author-page/index.js:5]: can remove this `props.data.stories;`

**Review by nandap4790 — APPROVED** [review]

**Pull request merged**